README.md

```diff
@@ -100,52 +169,75 @@ and from then on a simple
 
 is enough because the logger has been decorated with the request UUID already and from now on carries this information around.
 
+In other cases however, it might be useful to attach some meta-data to only one message and that can be achieved by using the `metadata` parameter from above:
+
+    logger.trace("user \(userID) logged in",
+                 metadata: ["extra_user_info": [ "favourite_colour": userFaveColour,
+                                                 "auth_method": "password" ]])
+
+The above invocation will log the message alongside meta-data merged from the `logger`'s meta-data and the meta-data provided with the `logger.trace` call.
+
+The logging meta-data is of type `typealias Logging.Metadata = [String: Metadata.Value]` where `Metadata.Value` is the following `enum` allowing nested structures rather than just `String`s as values.
+
+    public enum MetadataValue {
```

Comment:
weird nitpick; in text we say "meta-data" yet in names we do "Metadata" as if there was no "-".
May I propose to amend the text wording to be "metadata"?

Reply:
okay, will search & rep